File: ext/geometry/Rect.java
package geometry;

public class Rect {

  double x;
  double y;
  double width;
  double height;

  public Rect(double x, double y, double width, double height) {
    this.x = x;
    this.y = y;
    this.width = width;
    this.height = height;
  } 
 
  public boolean containsPoint(Point point) {
    return point.getX() >= this.x && point.getY() >= this.y &&
           point.getX() <= (this.x + this.width) && point.getY() <= (this.y + this.height);
  }

  public double getX() {
    return this.x;
  }
    
  public double getY() {
    return this.y;
  }
  
  public double getWidth() {
    return this.width;
  }

  public double getHeight() {
    return this.height;
  }
  
  @Override public boolean equals(Object other) {
    boolean result = false;
    if (other instanceof Rect) {
      Rect rect = (Rect) other;
      result = (this.x == rect.getX() && this.y == rect.getY() && 
        this.width == rect.getWidth() && this.height == rect.getHeight());
    }
    return result;
  }
  
}

File: lib/local_geocoder/geometry/rect.rb
module LocalGeocoder
  module Geometry

    Rect = Struct.new(:x, :y, :width, :height) do
      def contains_point?(point)
        point.x >= self.x && point.y >= self.y &&
          point.x <= (self.x+self.width) && point.y <= (self.y+self.height)
      end
    
      def ==(rect)
        self.x == rect.x && self.y == rect.y &&
          self.width == rect.width && self.height == rect.height
      end
    
      def inspect
        "[#{x},#{y}],[#{x+width},#{y+height}]"
      end
      
    end

  end
end
